src/main/java/org/ow2/proactive/catalogclient/service/CatalogObjectService.java

```diff
@@ -0,0 +1,145 @@
+            CATCH_URL_REGEX_WITH_HTML_QUOTE + "))\\)";
+
+    @Autowired
+    private RemoteObjectService remoteObjectService;
+
+    @Autowired
+    private ServiceConfiguration serviceConfiguration;
+
+    /**
+     * Get the metadata of a catalog object
+     * @param catalogURL is the catalog url
+     * @param bucketId is the bucket containing the object id
+     * @param name is the object name
+     * @return a Catalog Object containing the object information
+     */
+    public CatalogObject getCatalogObjectMetadata(String catalogURL, long bucketId, String name) {
+
+        final String url = getURL(catalogURL, bucketId, name, false);
+        return (CatalogObject) remoteObjectService.sendRequest(url, CatalogObject.class);
+    }
+
+    /**
+     * Get an object from the catalog
+     * @param catalogURL is the catalog url
+     * @param bucketId is the bucket containing the object id
+     * @param name is the object name
+     * @return a Catalog Object containing the object information
+     */
+    public String getRawCatalogObject(String catalogURL, long bucketId, String name) {
+
+        final String url = getURL(catalogURL, bucketId, name, true);
+        return (String) remoteObjectService.sendRequest(url, String.class);
+    }
+
+    /**
+     * Get a resource from the catalog and resolve PA:GET_FROM_URL("url") if necessary
+     * @param bucketId is the resource bucket id
+     * @param myResourceID is the resource name
+     * @param resolveLinks on true replace PA:GET_FROM_URL("url") by its value otherwise return the raw resource
+     * @return a string which contains the
+     */
+    public String getResource(long bucketId, String myResourceID, boolean resolveLinks) {
+
+        Pattern pattern = Pattern.compile(GET_FROM_URL_PATTERN);
+
+        String resource = getRawCatalogObject(serviceConfiguration.getCatalogURL(),
+                bucketId,
+                myResourceID);
+        if (!resolveLinks) {
+            return resource;
+        }
+        Matcher tokenMatcher = pattern.matcher(resource);
+        String resourceURL;
+        while (tokenMatcher.find()) {
+            resourceURL = extractURLFromToken(tokenMatcher.group());
+            resource = tokenMatcher.replaceFirst((String) remoteObjectService.sendRequest(resourceURL, String.class));
+            tokenMatcher.reset(resource);
+        }
+        return resource;
+    }
+
+    /**
+     * Generate the catalog URL from the argument
+     * @param catalogURL is the domain name
+     * @param bucketId is the id of bucket containing the resource
+     * @param name is the resource name
+     * @param raw enables to choose for the resource metadata or the resource content
+     * @return the resource content if raw is true otherwise return the resource metadata
+     */
+    @VisibleForTesting
+    String getURL(String catalogURL, long bucketId, String name, boolean raw) {
+
+        final String BUCKETS_PATH = "buckets/";
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS00117)